typescript/infra/src/core/deploy.ts

```diff
@@ -2,177 +2,205 @@ import { ethers } from 'ethers';
 import path from 'path';
 
 import {
-  AbacusConnectionManager,
   AbacusConnectionManager__factory,
-  Inbox,
   InboxValidatorManager,
   InboxValidatorManager__factory,
   Inbox__factory,
   InterchainGasPaymaster__factory,
-  OutboxValidatorManager,
   OutboxValidatorManager__factory,
   Outbox__factory,
-  UpgradeBeaconController,
   UpgradeBeaconController__factory,
 } from '@abacus-network/core';
 import { AbacusAppDeployer, ProxiedContract } from '@abacus-network/deploy';
 import {
   AbacusCore,
+  ChainMap,
   ChainName,
   CoreContractAddresses,
-  ProxiedAddress,
+  CoreContracts,
+  DomainConnection,
+  InboxContracts,
+  MailboxAddresses,
+  MultiProvider,
+  RemoteChainMap,
+  Remotes,
+  domains,
+  utils as sdkUtils,
 } from '@abacus-network/sdk';
 import { types } from '@abacus-network/utils';
 
 import { DeployEnvironment, RustConfig } from '../config';
 
-import { CoreConfig, ValidatorManagerConfig } from './types';
+import { ValidatorManagerConfig } from './types';
 
-export class AbacusCoreDeployer extends AbacusAppDeployer<
-  CoreContractAddresses,
-  CoreConfig
+export class AbacusCoreDeployer<
+  Networks extends ChainName,
+> extends AbacusAppDeployer<
+  Networks,
+  ValidatorManagerConfig,
+  CoreContractAddresses<Networks, any>
 > {
-  async deployContracts(
-    domain: types.Domain,
-    config: CoreConfig,
-  ): Promise<CoreContractAddresses> {
-    const overrides = this.getOverrides(domain);
-    const signer = this.mustGetSigner(domain);
-    const upgradeBeaconController: UpgradeBeaconController =
-      await this.deployContract(
-        domain,
-        'UpgradeBeaconController',
-        new UpgradeBeaconController__factory(signer),
-      );
-
-    const outboxValidatorManagerConfig = this.validatorManagerConfig(
-      config,
-      domain,
+  async deployContracts<Local extends Networks>(
+    network: Local,
+    config: ValidatorManagerConfig,
+  ): Promise<CoreContractAddresses<Networks, Local>> {
+    const dc = this.multiProvider.getDomainConnection(network);
+    const signer = dc.signer!;
```

Comment:
Can we really assume a signer here? IMO the multiprovider should have a similar function to `mustGetSigner`. Or if you feeling fp you can add the option monad :P

Reply:
I'm joking about the latter, at least in this PR :P

Reply:
afaik the `!` behaves the same as `mustGetSigner` without the nice error message
do you want me to add the message back?

Reply:
https://github.com/abacus-network/abacus-monorepo/issues/394